### PR TITLE
RGBD Odometry Pybinds

### PR DESCRIPTION
**Pull request by reyanshsolis**
---

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3359)

### CODE REVIEW

**Review by ssheorey — CHANGES_REQUESTED** [review]
> Reviewed 1 of 54 files at r2, 2 of 48 files at r3, 7 of 11 files at r9.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3359)* status: 10 of 67 files reviewed, 11 unresolved discussions (waiting on @reyanshsolis and @theNded)

___
*[cpp/pybind/t/geometry/tsdf_voxelgrid.cpp, line 36 at r3](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZobspI67VsTVxrK2Xo:-MZobspI67VsTVxrK2Xp:b-uyts1t) ([raw file](https://github.com/intel-isl/open3d/blob/91fb2cbee0b533138a82fe46a66caa4e6997c35c/cpp/pybind/t/geometry/tsdf_voxelgrid.cpp#L36)):*
> ```cpp
> namespace t {
> namespace geometry {
> ```

Add documentation (ReST format) for pybinds (including for arguments). Ensure documentation is equivalent to C++. 
___
*[cpp/pybind/t/geometry/tsdf_voxelgrid.cpp, line 81 at r3](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZohTlJAOXMFa-sjWUz:-MZohTlJAOXMFa-sjWV-:bfhr5fo) ([raw file](https://github.com/intel-isl/open3d/blob/91fb2cbee0b533138a82fe46a66caa4e6997c35c/cpp/pybind/t/geometry/tsdf_voxelgrid.cpp#L81)):*
> ```cpp
>             "depth_scale"_a, "depth_max"_a);
> 
>     // TODO(wei): expose mask code as a python class
> ```

Is this done already? I can see the docs.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 47 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoit6kCFvNfzgdiGQU:-MZoit6kCFvNfzgdiGQV:b-g921p1) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L47)):*
> ```cpp
>     py::class_<OdometryConvergenceCriteria> odometry_convergence_criteria(
>             m, "OdometryConvergenceCriteria",
>             "Class that defines the convergence criteria of odometry. "
> ```

~~Class that defines the~~ convergence criteria of odometry.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 74 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoia9M6S26E_G7juXD:-MZoia9M6S26E_G7juXE:bbt825h) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L74)):*
> ```cpp
>                         "with max_iteration={:d}, relative_rmse={:e}, "
>                         "relative_fitness={:e}.",
>                         c.max_iteration_, c.relative_rmse_,
> ```

Shorten: 
`"ICPConvergenceCriteria[max_iteration={:d}, relative_rmse={:e}, relative_fitness={:e}."]`
Similarly for other `__repr__` output.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 109 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZokQNN8eUYVlVZxkrT:-MZokQNN8eUYVlVZxkrU:bqayyly) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L109)):*
> ```cpp
>     py::class_<OdometryLossParams> odometry_loss_params(
>             m, "OdometryLossParams",
>             "Class that contains the odometry loss parameters.");
> ```

~~Class that contains~~ the odometry loss parameters
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 161 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoknJN8ROnTUgX7wvL:-MZoknJN8ROnTUgX7wvM:btprpnf) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L161)):*
> ```
>                  "(H, W, 1) Float32 source depth image obtained by "
> ```
switch to (rows,cols, channels=1) here and similarly below.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 164 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZol2aD3u4OMcSQjZgg:-MZol2aD3u4OMcSQjZgh:b669p8s) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L164)):*
> ```
> depth gradient image at x-axis
> ```
at -> along the 
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 208 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZolIV272gmFyf7BUBo:-MZolIV272gmFyf7BUBp:b-baz9l5) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L208)):*
> ```cpp
>                                                         core::Device("CPU:0")),
>           "depth_scale"_a = 1000.0f, "depth_max"_a = 3.0f,
>           "criterias"_a = std::vector<OdometryConvergenceCriteria>({10, 5, 3}),
> ```

criterias -> criteria is plural
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 221 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZolZXa3y-HGSvv_1t0:-MZolZXa3y-HGSvv_1t1:bkfpqp7) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L221)):*
<details><summary><i alt="summary">Quoted 5 lines of code…</i></summary>

> ```
>           "[(V_p - V_q)^T N_p]^2, where "
>           "V_p  denotes the vertex at pixel p in the source, "
>           "V_q  denotes the vertex at pixel q in the target, "
>           "N_p  denotes the normal at pixel p in the source. "
>           "q is obtained by transforming p with init_source_to_target then "
> ```
</details>

Use `:math:` ReST markup.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 237 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoljxo5OhihY_iWF1V:-MZoljxo5OhihY_iWF1W:b-ezyl6u) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L237)):*
<details><summary><i alt="summary">Quoted 5 lines of code…</i></summary>

> ```
>           "(I_p - I_q)^2, where "
>           "I_p  denotes the intensity at pixel p in the source, "
>           "I_q  denotes the intensity at pixel q in the target. "
>           "q is obtained by transforming p with init_source_to_target then "
>           "projecting with intrinsics. "
> ```
</details>

Use `:math:` ReST markup.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 256 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZolnki5oUCjoCoWCfN:-MZolnki5oUCjoCoWCfO:b5m8crs) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L256)):*
<details><summary><i alt="summary">Quoted 6 lines of code…</i></summary>

> ```
>           "(I_p - I_q)^2 + lambda(D_p - (D_q)')^2, where "
>           "I_p  denotes the intensity at pixel p in the source, "
>           "I_q  denotes the intensity at pixel q in the target. "
>           "D_p  denotes the depth pixel p in the source, "
>           "D_q  denotes the depth pixel q in the target. "
>           "q is obtained by transforming p with init_source_to_target then "
> ```
</details>

Use `:math:` ReST markup.

**Review by theNded — APPROVED** [review]
Reviewable keeps asking me to review 150+ files so use GitHub default...
After minor document changes are resolved we can merge.

File: cpp/open3d/t/pipelines/odometry/RGBDOdometry.h [hunk 1]
```diff
@@ -263,19 +275,19 @@ OdometryResult ComputeOdometryResultIntensity(
 /// image obtained by RGBToGray before calling this function.
 /// \param target_intensity (rows, cols, channels=1) Float32 target intensity
 /// image obtained by RGBToGray before calling this function.
-/// \param source_depth_dx (rows, cols, channels=1) Float32 source depth
-/// gradient image at x-axis obtained by FilterSobel before calling this
+/// \param target_depth_dx (rows, cols, channels=1) Float32 target depth
```
Comment (reyanshsolis) [inline cpp/open3d/t/pipelines/odometry/RGBDOdometry.h:278]: @theNded The `source_*` and `target_*` might have been interchanged in this file. I found them to be opposite in `.cpp`, may you please verify this in the new changes, if they are fixed?

File: cpp/pybind/t/pipelines/odometry/odometry.cpp [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+          "method"_a = Method::Hybrid, "params"_a = OdometryLossParams());
+    docstring::FunctionDocInject(m, "rgbd_odometry_multi_scale",
+                                 map_shared_argument_docstrings);
+
+    m.def("compute_odometry_result_point_to_plane",
+          &ComputeOdometryResultPointToPlane,
+          "(Estimates the OdometryResult (4x4 rigid transformation "
+          "T from source to target, with inlier rmse and fitness). "
+          "Performs one iteration of RGBD odometry using "
+          "Loss function: :math: `[(V_p - V_q)^T N_p]^2`, where "
```
Comment (reyanshsolis) [inline cpp/pybind/t/pipelines/odometry/odometry.cpp:211]: @ssheorey I tried, but it doesn't seem to work. May you please help me with this. 
I referred to the legacy `odometry` and `trainglemesh` pybinds, but it is still not fixed.

File: cpp/open3d/t/pipelines/registration/Registration.h [hunk 3]
```diff
@@ -162,17 +161,16 @@ RegistrationResult RegistrationICP(
 /// \param max_correspondence_distances VectorDouble of maximum correspondence
 /// points-pair distances of type double, for each iteration. Must be of same
 /// length as voxel_sizes and criterias.
-/// \param init Initial transformation estimation of type Float64 on CPU.
-/// \param estimation Estimation method.
+/// \param init_source_to_target Initial transformation estimation of type
+/// Float64 on CPU. \param estimation Estimation method.
```
[the remark this answers is not in the document]
Comment (reyanshsolis) [reply cpp/open3d/t/pipelines/registration/Registration.h]: Done.

File: cpp/open3d/t/pipelines/registration/Registration.h [hunk 4]
```diff
@@ -131,16 +131,15 @@ RegistrationResult EvaluateRegistration(
 /// \param target The target point cloud.
 /// \param max_correspondence_distance Maximum correspondence points-pair
 /// distance.
-/// \param init Initial transformation estimation of type Float64 on CPU.
-/// \param estimation Estimation method.
-/// \param criteria Convergence criteria.
+/// \param init_source_to_target Initial transformation estimation of type
+/// Float64 on CPU. \param estimation Estimation method. \param criteria
```
[the remark this answers is not in the document]
Comment (reyanshsolis) [reply cpp/open3d/t/pipelines/registration/Registration.h]: Done.

File: cpp/pybind/t/pipelines/odometry/odometry.cpp [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                {"intrinsics", "(3, 3) intrinsic matrix for projection."},
+                {"intensity_huber_delta",
+                 "Huber norm parameter used in intensity loss."},
+                {"method",
+                 "Estimation method used to apply RGBD odometry. "
+                 "One of (``PointToPlane``, ``Intensity``, ``Hybrid``)"},
+                {"params", "Odometry loss parameters."},
+                {"source", "The source RGBD image."},
+                {"source_depth",
+                 "(H, W, 1) Float32 source depth image obtained by "
```
[the remark this answers is not in the document]
Comment (reyanshsolis) [reply cpp/pybind/t/pipelines/odometry/odometry.cpp]: Done.

File: cpp/pybind/t/pipelines/odometry/odometry.cpp [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+          "criterias"_a = std::vector<OdometryConvergenceCriteria>({10, 5, 3}),
+          "method"_a = Method::Hybrid, "params"_a = OdometryLossParams());
+    docstring::FunctionDocInject(m, "rgbd_odometry_multi_scale",
+                                 map_shared_argument_docstrings);
+
+    m.def("compute_odometry_result_point_to_plane",
+          &ComputeOdometryResultPointToPlane,
+          "Estimates the 4x4 rigid transformation T from source to target. "
+          "Performs one iteration of RGBD odometry using loss function "
+          "[(V_p - V_q)^T N_p]^2, where "
```
[the remark this answers is not in the document]
Comment (reyanshsolis) [reply cpp/pybind/t/pipelines/odometry/odometry.cpp]: I found some `:math:` references, but it is still not working as expected. Will try to fix it soon.

**Review by reyanshsolis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3359)* status: 7 of 67 files reviewed, 17 unresolved discussions (waiting on @reyanshsolis, @ssheorey, @theNded, and @yxlao)

___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 47 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoit6kCFvNfzgdiGQU:-M_0PrHP2OY3VEeEt3Wj:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L47)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

~~Class that defines the~~ convergence criteria of odometry.
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 74 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoia9M6S26E_G7juXD:-M_0Ps0AC6XDq53a8ZB2:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L74)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

Shorten: 
`"ICPConvergenceCriteria[max_iteration={:d}, relative_rmse={:e}, relative_fitness={:e}."]`
Similarly for other `__repr__` output.
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 109 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZokQNN8eUYVlVZxkrT:-M_0PsRi28GEwWRZHP3J:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L109)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

~~Class that contains~~ the odometry loss parameters
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 161 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoknJN8ROnTUgX7wvL:-M_0Pt0w7dkBMKS6P-t7:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L161)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

> ```
>                  "(H, W, 1) Float32 source depth image obtained by "
> ```
switch to (rows,cols, channels=1) here and similarly below.
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 164 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZol2aD3u4OMcSQjZgg:-M_0PtN3DgdLfvbeNXdo:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L164)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

> ```
> depth gradient image at x-axis
> ```
at -> along the 
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 208 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZolIV272gmFyf7BUBo:-M_0PtzBD4nV2CEZGE1S:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L208)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

criterias -> criteria is plural
</blockquote></details>

Done.

File: cpp/pybind/t/pipelines/odometry/odometry.cpp (hunk 2 above)
Comment (reyanshsolis) [reply cpp/pybind/t/pipelines/odometry/odometry.cpp]: Done.

File: cpp/pybind/t/pipelines/odometry/odometry.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (reyanshsolis) [reply cpp/pybind/t/pipelines/odometry/odometry.cpp]: Done.

**Review by reyanshsolis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3359)* status: 7 of 68 files reviewed, 16 unresolved discussions (waiting on @reyanshsolis, @ssheorey, @theNded, and @yxlao)

___
*[cpp/pybind/t/geometry/tsdf_voxelgrid.cpp, line 36 at r3](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZobspI67VsTVxrK2Xo:-M_3_QbS9HIhmrj0diPw:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/91fb2cbee0b533138a82fe46a66caa4e6997c35c/cpp/pybind/t/geometry/tsdf_voxelgrid.cpp#L36)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

Add documentation (ReST format) for pybinds (including for arguments). Ensure documentation is equivalent to C++. 
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 221 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZolZXa3y-HGSvv_1t0:-M_3_W9mC3SZUp_RpZzE:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L221)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

> ```
>           "[(V_p - V_q)^T N_p]^2, where "
>           "V_p  denotes the vertex at pixel p in the source, "
>           "V_q  denotes the vertex at pixel q in the target, "
>           "N_p  denotes the normal at pixel p in the source. "
>           "q is obtained by transforming p with init_source_to_target then "
> ```
Use `:math:` ReST markup.
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 237 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZoljxo5OhihY_iWF1V:-M_3_Wmx-bJZjPGJ3l6R:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L237)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

> ```
>           "(I_p - I_q)^2, where "
>           "I_p  denotes the intensity at pixel p in the source, "
>           "I_q  denotes the intensity at pixel q in the target. "
>           "q is obtained by transforming p with init_source_to_target then "
>           "projecting with intrinsics. "
> ```
Use `:math:` ReST markup.
</blockquote></details>

Done.
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 256 at r9](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZolnki5oUCjoCoWCfN:-M_3_XGOC-ZRe60sXUG0:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/a458ddc8f3801e68ba2e28f00b6e16b73ee3689c/cpp/pybind/t/pipelines/odometry/odometry.cpp#L256)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

> ```
>           "(I_p - I_q)^2 + lambda(D_p - (D_q)')^2, where "
>           "I_p  denotes the intensity at pixel p in the source, "
>           "I_q  denotes the intensity at pixel q in the target. "
>           "D_p  denotes the depth pixel p in the source, "
>           "D_q  denotes the depth pixel q in the target. "
>           "q is obtained by transforming p with init_source_to_target then "
> ```
Use `:math:` ReST markup.
</blockquote></details>

Done.

**Review by theNded — APPROVED** [review]

**Review by theNded — APPROVED** [review]

File: cpp/open3d/t/pipelines/odometry/RGBDOdometry.h (hunk 1 above)
Comment (reyanshsolis) [reply cpp/open3d/t/pipelines/odometry/RGBDOdometry.h:278]: Done.

**Review by reyanshsolis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3359)* status: 4 of 68 files reviewed, 8 unresolved discussions

___
*[cpp/pybind/t/geometry/tsdf_voxelgrid.cpp, line 36 at r3](https://reviewable.io/reviews/intel-isl/open3d/3359#-MZobspI67VsTVxrK2Xo:-M_9Ocbh3wdsuTDg-Ns9:b1by660) ([raw file](https://github.com/intel-isl/open3d/blob/91fb2cbee0b533138a82fe46a66caa4e6997c35c/cpp/pybind/t/geometry/tsdf_voxelgrid.cpp#L36)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

Hey, sorry, I thought this file was part of this PR. But this file is missing Python docs, so we should add them in some PR.
</blockquote></details>

Sure, I will do that soon in some other PR. Sorry, I thought it was referring to Odometry `ReST` changes, so I marked it complete. 
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 54 at r14](https://reviewable.io/reviews/intel-isl/open3d/3359#-M_7dVICAtRvR82S_EAV:-M_9PZCW7tyqPmC7auPB:b-etnjrt) ([raw file](https://github.com/intel-isl/open3d/blob/43a6055cee4956a8cdeb296d1100516012de92b3/cpp/pybind/t/pipelines/odometry/odometry.cpp#L54)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

> ```
> def(py::init
> ```
Add ClassMethodDocInject for `__init__` functions of all classes.
</blockquote></details>

Sorry, I missed the comment before merging, I will add it along with one of my other PR. 
___
*[cpp/pybind/t/pipelines/odometry/odometry.cpp, line 194 at r14](https://reviewable.io/reviews/intel-isl/open3d/3359#-M_7dkDWBjdIjSYa0rdv:-M_9P12_3p1Yo01UkrVA:bnt6dhh) ([raw file](https://github.com/intel-isl/open3d/blob/43a6055cee4956a8cdeb296d1100516012de92b3/cpp/pybind/t/pipelines/odometry/odometry.cpp#L194)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

Can we release GIL here (and in other odometry methods) to enable Python multithreading parallelism? This looks like a long running compute bound function. [Also @thended ] I don't believe we use any Python interpreter functionality inside.
</blockquote></details>

I am unfamiliar with GIL, it would be great, if you may share some references. I will try to apply that in `Odometry` and `Registration` as well in a separate PR.

**Pull request merged**